src/Aspire.Hosting.Python/PythonProjectResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Python;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Python applications to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class PythonProjectResourceBuilderExtensions
+{
+    /// <summary>
+    /// Adds a python application with a virtual environment to the application model.
+    /// The Python project should have a virtual environment set up in the project directory under the name ".venv".
+    /// <para>
+    /// The python script in the project is automatically instrumented with opentelemetry when the virtual environment
+    /// contains the opentelemetry-instrument executable. You can get this by adding the opentelemtry-distro package
+    /// to your python project. In addition to the opentelemetry-distro package, you need to add the opentelemetry-exporter-otlp
+    /// for the traces, logs, and metrics to be exported to the Aspire observability platform.
+    /// </para>
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="projectDirectory">The path to the directory containing the python project files.</param>
+    /// <param name="scriptPath">The path to the script relative to the project directory to run.</param>
+    /// <param name="virtualEnvironmentPath">Path to the virtual environment.</param>
+    /// <param name="scriptArgs">The arguments for the script.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<PythonProjectResource> AddPythonProject(
+        this IDistributedApplicationBuilder builder, string name, string projectDirectory, string scriptPath, string virtualEnvironmentPath = ".venv", params string[] scriptArgs)
```

Comment:
Can you make the default value for `virtualEnvironmentPath` `null` and then check for it and replace it with the default inside the method. Also note the earlier comments I made about making `<summary>` shorter and more concise and adding a `<remarks>` section (where you can have your paragraphs.